rfcs/rdm-0064-custom-fields.md

```diff
@@ -0,0 +1,614 @@
+
+- :white_check_mark: It is available in InvenioRDM v10
+- :x: It is **not** available in InvenioRDM v10
+- :grey_question: It is available in InvenioRDM v10, but with some limitations.
+
+**As an instance administrator...**
+
+- :white_check_mark: I want to add a custom fields to the record's data model, and/or the community's data model.
+- :white_check_mark: I want users to be able to input values for those custom fields via the upload and community creation forms.
+- :grey_question: I want to add custom fields for internal use only (i.e. not visible to the users, nor available in the forms).
+- :x: I want to be able to hide some fields in some communities. While the fields might still be present in the data model, they won't be shown/available for input. Alternatively, I want to be able to configure custom fields per community.
+
+**The following use cases apply to all the above points**, and focus on the how fields are defined and used.
+
+- :grey_question: I want to be able to define the type: primitive (string, number, boolean, date), complex (list, dictionary), and vocabularies (both generic and specific).
+- :x: I want to be able to define required fields.
+
+**When displaying the custom fields in the forms...**
+
+- :white_check_mark: I want to define groups for my new fields.
+- :white_check_mark: I want to define in which order they will display inside those groups.
+- :x: I want to add fields to already existing groups (e.g. Basic information).
+- :white_check_mark: I want to support translations (i18n) of the fields' label and description.
+
+**When displaying the custom fields of records...**
+
+- :white_check_mark: I want to define where and how the field will be shown in the landing/community page.
+- :white_check_mark: I want to define how the values will be serialized into different formats (e.g. Dublin Core, Darwin Core, DataCite).
+- :white_check_mark: I want to be able to filter my search results based on my new field.
+- :white_check_mark: I want to support translations (i18n) for the values of the custom fields, when using vocabularies.
+
+### Future work
+
+This RFC targets the implementation of two field types, allowing to add primitive types custom fields (e.g. free text, boolean, number) and also based on generic vocabularies (e.g. defining a new "experiment" vocabulary). This will enable the implementation of the mechanisms to support custom fields, and allow developers to impelement new types of fields (e.g. dates).
+
+Functionalities that might come in the future:
+
+- Configuration of custom fields per community. Meaning, each community has its own specific custom fields.
+- Internal use field, that will not be visible by end users in the UI nor in the REST API.
+- Applying permissions per-field.
+- Configuring custom fields using the UI, for example the _Backoffice_.
+- Compatibility with OpenSearch.
+- Add custom fields to existing sections, for example _Basic Information_.
+
+## Detailed design
+
```

Comment:
optional: add challenges, referencing to the general architecture (resource, service, data layer)?